### PR TITLE
Clarify duplex semantics

### LINKED ISSUE #741: Semantics: full-duplex related semantics ...
[7.5](https://www.ietf.org/archive/id/draft-ietf-httpbis-semantics-14.html#name-response-correlation) briefly touches this topic.

> Responses (both final and interim) can be sent at any time after a request is received, even if it is not yet complete

More explicit descriptions are helpful, such as:

1. responses can be sent at any time, successful for failed ones
2. responses may be completed before requests are completed. This is described for [100-continue](https://www.ietf.org/archive/id/draft-ietf-httpbis-semantics-14.html#name-expect) but should not be limited to 100-continue. 
3. for 2), should a client that sees a failure response continues sending the remaining request content?

### PR DESCRIPTION
**Pull request by mnot**
Fixes #741

### CODE REVIEW

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -2681,12 +2681,18 @@ Host: www.example.org
    messages, while others use an explicit identifier.
 </t>
 <t>
-   Responses (both <x:ref>final</x:ref> and <x:ref>interim</x:ref>) can be sent
-   at any time after a
+   All responses, regardless of the status code (including <x:ref>interim</x:ref>
+   responses) can be sent (and be completed) at any time after a
```
Comment (martinthomson) [inline draft-ietf-httpbis-semantics-latest.xml:2685]: Lots of parentheses here.  Not sure what to do about that, as they are all useful.  I would probably just use more sentences 🤷‍♂️

**Review by martinthomson — APPROVED** [review]

**Review by royfielding — CHANGES_REQUESTED** [review]
This reads odd to me. I was about to say the same as Martin. It would be better to explain each case and why we care. But no specific suggestion in mind yet.

**Comment by mnot** [comment]
I saw that too, but favoured compactness over avoiding parenthesis. Happy to take suggestions.

**Comment by mnot** [comment]
Try that.

File: _config.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
   title: HTTP Core
   description: Revising the documents that define HTTP
-  remote_theme: httpwg/theme
+  remote_theme: httpwg/theme@main
```
Comment (martinthomson) [inline _config.yml:4]: ?

File: draft-ietf-httpbis-semantics-latest.xml [hunk 3]
```diff
@@ -2681,12 +2681,19 @@ Host: www.example.org
    messages, while others use an explicit identifier.
 </t>
 <t>
-   Responses (both <x:ref>final</x:ref> and <x:ref>interim</x:ref>) can be sent
-   at any time after a
-   request is received, even if it is not yet complete. However, clients
+   All responses, regardless of the status code (including <x:ref>interim</x:ref>
+   responses) can be sent at any time after a
+   request is received, even if it is not yet complete. This applies even if the
+   response completes before the request does. However, clients
```
Comment (martinthomson) [inline draft-ietf-httpbis-semantics-latest.xml:2687]:
```suggestion
   request is received, even if it is not yet complete. A response can complete
   before the request does. Clients
```

**Review by martinthomson — APPROVED** [review]

File: _config.yml (hunk 2 above)
Comment (mnot) [reply _config.yml:4]: wrong branch.

**Review by reschke — APPROVED** [review]

File: draft-ietf-httpbis-semantics-latest.xml [hunk 4]
```diff
@@ -2681,12 +2681,19 @@ Host: www.example.org
    messages, while others use an explicit identifier.
 </t>
 <t>
-   Responses (both <x:ref>final</x:ref> and <x:ref>interim</x:ref>) can be sent
-   at any time after a
-   request is received, even if it is not yet complete. However, clients
+   All responses, regardless of the status code (including <x:ref>interim</x:ref>
+   responses) can be sent at any time after a
+   request is received, even if it is not yet complete. A response can complete
+   before the request does. Clients
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:2687]:
```suggestion
   All responses, regardless of the status code (including <x:ref>interim</x:ref>
   responses) can be sent at any time after a
   request is received, even if the request is not yet complete. A response can complete
   before its corresponding request is complete. Clients
```

File: draft-ietf-httpbis-semantics-latest.xml [hunk 5]
```diff
@@ -2681,12 +2681,19 @@ Host: www.example.org
    messages, while others use an explicit identifier.
 </t>
 <t>
-   Responses (both <x:ref>final</x:ref> and <x:ref>interim</x:ref>) can be sent
-   at any time after a
-   request is received, even if it is not yet complete. However, clients
+   All responses, regardless of the status code (including <x:ref>interim</x:ref>
+   responses) can be sent at any time after a
+   request is received, even if the request is not yet complete. A response can complete
+   before its corresponding request is complete. Clients
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:2687]:
```suggestion
   All responses, regardless of the status code (including <x:ref>interim</x:ref>
   responses) can be sent at any time after a request is received, even if the
   request is not yet complete. A response can complete before its
   corresponding request is complete. Clients
```

File: draft-ietf-httpbis-semantics-latest.xml [hunk 6]
```diff
@@ -2681,12 +2681,19 @@ Host: www.example.org
    messages, while others use an explicit identifier.
 </t>
 <t>
-   Responses (both <x:ref>final</x:ref> and <x:ref>interim</x:ref>) can be sent
-   at any time after a
-   request is received, even if it is not yet complete. However, clients
+   All responses, regardless of the status code (including <x:ref>interim</x:ref>
+   responses) can be sent at any time after a request is received, even if the
+   request is not yet complete. A response can complete before its
+   corresponding request is complete. Clients
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:2687]:
```suggestion
   All responses, regardless of the status code (including <x:ref>interim</x:ref>
   responses) can be sent at any time after a request is received, even if the
   request is not yet complete. A response can complete before its
   corresponding request is complete. Likewise, clients are not expected
   to wait any specific amount of time for a response. Clients
```

**Review by royfielding — APPROVED** [review]
Okay, I have tweaked this editorially so that the paragraph makes sense, IMO.

**Review by reschke — APPROVED** [review]

**Pull request merged**